LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/LoRaDeviceAPIService.cs

```diff
@@ -176,17 +177,22 @@ private async Task<SearchDevicesResult> SearchByEuiAsync(string eui)
             {
                 if (response.StatusCode == System.Net.HttpStatusCode.NotFound)
                 {
-                    return new SearchDevicesResult();
+                    return default;
                 }
 
                 this.logger.LogError($"error calling get device/station by EUI api: {response.ReasonPhrase}, status: {response.StatusCode}, check the azure function log");
 
-                return new SearchDevicesResult();
+                return default;
             }
 
             var result = await response.Content.ReadAsStringAsync();
-            var devices = (List<IoTHubDeviceInfo>)JsonConvert.DeserializeObject(result, typeof(List<IoTHubDeviceInfo>));
-            return new SearchDevicesResult(devices);
+
+            if (string.IsNullOrEmpty(result))
+                return null;
+
+            return JsonDocument.Parse(result).RootElement is { ValueKind: JsonValueKind.Object } someObject
+                ? someObject.TryGetProperty("PrimaryKey", out var propertyValue) ? propertyValue.GetString() : null
+                : null;
```

Comment:
nit: Perhaps go one more step further here and make sure that `PrimaryKey` is a string?

```suggestion
            return result is { Length: > 0 } json
                   && JsonDocument.Parse(json).RootElement is { ValueKind: JsonValueKind.Object } someObject
                   && someObject.TryGetProperty("PrimaryKey", out var propertyValue)
                   && propertyValue is { ValueKind: JsonValueKind.String }
                 ? propertyValue.GetString()
                 : null;
```